### DISCUSSION TITLE
sorry!when can we use this project as a golang library?[feature]

### DISCUSSION THREAD

**Discussion by Nyx2022** [projectdiscovery/dnsx#89] [Q&A] [answered]
**As this title**
I wanna know that when can we use dnsx as project subfinder's EnumerateSingleDomain method in our project?maybe i'm too anxious

**Comment by Mzack9999** [comment, marked as answer]
@Nyx2020 You can use the `dnsx` library directly through the `New` method https://github.com/projectdiscovery/dnsx/blob/b4dc97030b21278f4b3d3bb7be85e87538f10b49/libs/dnsx/dnsx.go#L44 and perform all the various DNS operations that the library currently supports. Subfinder already uses dnsx internally to perform resolutions.
